File: cpp/55d_1506A_StrangeTable.cpp
#include<bits/stdc++.h>

#define ll unsigned long long

ll solve(ll n, ll m, ll x){
    int col = (x-1) / n , row = (x - 1) % n;
    return m * (row) + (col + 1);
}

int main(){
    ll t, n, m, x;
    std::vector<ll> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n>>m>>x;
        ans.push_back(solve(n,m,x));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/55d_1506A_StrangeTable.py
def solve(n, m, x):
    col, row = (x-1) // n, (x - 1) % n
    #print(row, col)
    return m * (row) + (col + 1)

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, m, x = input().split()
        n, m, x = int(n), int(m), int(x)
        ans.append(solve(n,m,x))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()